sdk/deviceupdate/azure-iot-deviceupdate/config.yaml

```diff
@@ -0,0 +1,29 @@
+input-file: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/deviceupdate/data-plane/Microsoft.DeviceUpdate/stable/2022-10-01/deviceupdate.json 
+java: true 
+output-folder: ./ 
+enable-sync-stack: true 
+generate-tests: true 
+regenerate-pom: false 
+title: DeviceUpdateClient 
+generate-sync-async-clients: true 
+generate-client-as-impl: true 
+generate-client-interfaces: false 
+service-interface-as-public: true 
+add-context-parameter: true 
+artifact-id: azure-iot-deviceupdate 
+data-plane: true 
+sync-methods: all 
+generate-samples: true 
+license-header: MICROSOFT_MIT_SMALL 
+namespace: com.azure.iot.deviceupdate 
+service-name: DeviceUpdate 
+context-client-method-parameter: true
+azure-arm: false 
+credential-types: tokencredential
+credential-scopes: https://api.adu.microsoft.com/.default 
+service-versions: 
+  - '2022-10-01' 
+polling: 
+    default: 
+        strategy: >- 
+                  new OperationResourcePollingStrategyWithEndpoint<>({httpPipeline}, "https://" + this.client.getEndpoint(), null, null, {context})
```

Comment:
We should be able to get rid of this `OperationResourcePollingStrategyWithEndpoint`, as the one from azure-core has supported this (`operation-location` header without endpoint).

Also can delete the `OperationResourcePollingStrategyWithEndpoint` class. https://github.com/Azure/azure-sdk-for-java/blob/dae1ec3e8502430d263b6b579e87279337f0f1ec/sdk/deviceupdate/azure-iot-deviceupdate/src/main/java/com/azure/iot/deviceupdate/implementation/OperationResourcePollingStrategyWithEndpoint.java